app/src/main/java/com/example/garden_planner/models/PlantInBed.java

```diff
@@ -21,4 +22,9 @@ public class PlantInBed extends ParseObject {
     public Plant getPlantType(){return (Plant)get(KEY_TYPE);}
 
     public void setPlantType(Plant plantType){put(KEY_TYPE, plantType);}
+
+    public String getThisPlantName(){return getString(KEY_NAME);}
```

Comment:
is the "This" necessary? I might just not know the naming conventions they taught you.

Reply:
I could call it something else - I just said "this" because it's the name of that specific plant, not the overall plant type name. what would you call it?

Reply:
Because this is an attribute of a specific object instance, the "This" is redundant. For example, you might create a new PlantInBec object and store it in variable `currentPlant`. Then when getting the plant name, you would be calling `currentPlant.getThisPlantName`, which doesn't provide any more info than `currentPlant.getPlantName`

In different coding languages, "this" has very specific meaning, usually referring to "this" specific instance of the object. So adding "this" to the method name might be a bit confusing depending on the context.

Reply:
Unless, you plan on allowing for custom naming for plants, like "Bob the tomato plant". In which case I would use something like, `getCustomPlantName` or `getDisplayName`. And use `getPlantType` or `getPlantSpecies` for the name of the plant type.